### ISSUE TITLE
CDSSM output layer for classification

### ISSUE THREAD

**Issue by hexiang129** [keras-team/keras#11292]
In CDSSM model implementation, it first calculate cosine similarity with Dot, which is a scalar, then it feeds the scalar to Dense with output size=2 and activation type being "softmax". This is really weird. How could one product a softmax for binary classification from a single scalar? Did I miss something here?

**Comment by hexiang129** [comment]
sorry, filed in the wrong place.
